### PR TITLE
Refactor using pandas

### PR DESCRIPTION
**Pull request by fu**

### CODE REVIEW

**Comment by fu** [comment]
Before we merge this, we need to fix 10 tests in ursgal2 - help ! :)

**Comment by fu** [comment]
It works now.

File: unimod_mapper/unimod_mapper.py [hunk 1]
```diff
@@ -90,6 +107,212 @@ def mapper(self, mapper):
...
+            "elements"
+        ].tolist()
+
+    def mass_to_names(self, mass, decimals=0):
+        lower_mass, upper_mass = self._determine_mass_range(mass, decimals=decimals)
+        return self.df.query("@lower_mass <= `mono_mass` <= @upper_mass")[
+            "Name"
+        ].unique()
+
+    def mass_to_combos(self, mass, n=2, decimals=3):
```
Comment (MKoesters) [inline unimod_mapper/unimod_mapper.py:198]: docstring would be helpful

File: unimod_mapper/unimod_mapper.py [hunk 2]
```diff
@@ -90,6 +107,212 @@ def mapper(self, mapper):
...
+        return self.df.query("`Accession` == @id")["mono_mass"].to_list()
+
+    def id_to_composition(self, id):
+        return self.df.query("`Accession` == @id")["elements"].to_list()
+
+    def id_to_name(self, id):
+        return self.df.query("`Accession` == @id")["Name"].to_list()
+
+    def _determine_mass_range(self, mass, decimals=0):
+        fraction = 1 / 10 ** (decimals + 1)
```
Comment (MKoesters) [inline unimod_mapper/unimod_mapper.py:175]: What are these magic numbers?

**Review by MKoesters — CHANGES_REQUESTED** [review]

**Pull request merged**